src/warehouse/index.js

```diff
@@ -567,6 +569,80 @@ function processWarehouseMessage(message, options) {
 
   // store columnTypes as each column is set, so as not to call getDataType again
   switch (eventType) {
+    case 'extract': {
+      // set properties common to both tracks and event table
+      const commonProps = {};
+      const commonColumnTypes = {};
+
+      setDataFromInputAndComputeColumnTypes(
+        utils,
+        eventType,
+        commonProps,
+        message.context,
+        commonColumnTypes,
+        options,
+        'context_',
+      );
+
+      // set event column based on event_text in the tracks table
+      const eventColName = utils.safeColumnName(options, 'event');
+      commonProps[eventColName] = utils.transformTableName(
+        options,
+        message[utils.safeColumnName(options, 'event')],
+      );
+      commonColumnTypes[eventColName] = 'string';
+
+      // -----start: event table------
+
+      // do not create event table in case of empty event name (after utils.transformColumnName)
+      if (_.toString(commonProps[eventColName]).trim() === '') {
+        break;
+      }
```

Comment:
Earlier we used this for tracks + events table compatibility. Do we need this for extract as well?

Reply:
We should probably return error right incase of extract?

Reply:
You are correct. I have raised an `InstrumentationError` in case the event name is missing from the payload.